diff -Naur itk4.0.0_orig/configure itk4.0.0/configure
--- itk4.0.0_orig/configure	2013-02-03 08:25:51.000000000 -0800
+++ itk4.0.0/configure	2013-11-09 11:03:37.185416415 -0800
@@ -5283,7 +5283,7 @@
 
 
 
-    vars="generic/itk.h generic/itkInt.h"
+    vars="generic/itk.h generic/itkInt.h generic/itkDecls.h"
     for i in $vars; do
 	# check for existence, be strict because it is installed
 	if test ! -f "${srcdir}/$i" ; then
diff -Naur itk4.0.0_orig/configure.in itk4.0.0/configure.in
--- itk4.0.0_orig/configure.in	2013-02-03 08:25:51.000000000 -0800
+++ itk4.0.0/configure.in	2013-11-09 11:03:11.281906189 -0800
@@ -87,7 +87,7 @@
 	        itkUtil.c
                 itkStubInit.c
                 ])
-TEA_ADD_HEADERS([generic/itk.h generic/itkInt.h])
+TEA_ADD_HEADERS([generic/itk.h generic/itkInt.h generic/itkDecls.h])
 TEA_ADD_INCLUDES([${ITCL_INCLUDES} -I\"`${CYGPATH} ${srcdir}/generic`\"])
 TEA_ADD_LIBS([])
 TEA_ADD_CFLAGS([])
