Description: Fix D examples
Author: Andrew Ross <andrewross@users.sourceforge.net>
Origin: upstream
Index: plplot-5.9.10/examples/d/x20d.d
===================================================================
--- plplot-5.9.10.orig/examples/d/x20d.d	2013-10-02 10:27:39.162480034 +0100
+++ plplot-5.9.10/examples/d/x20d.d	2013-10-02 10:29:03.410477825 +0100
@@ -6,6 +6,7 @@
 
 import std.math;
 import std.stdio;
+import std.file;
 
 import plplot;
 
