Description: Upstream patch to fix wxwidgets linkage.
Author: Andrew Ross <andrewross@users.sourceforge.net>
Last-Update: 2014-07-01
Index: plplot-5.10.0/bindings/wxwidgets/CMakeLists.txt
===================================================================
--- plplot-5.10.0.orig/bindings/wxwidgets/CMakeLists.txt	2013-12-19 01:35:00.000000000 +0000
+++ plplot-5.10.0/bindings/wxwidgets/CMakeLists.txt	2014-07-01 12:25:05.556785342 +0100
@@ -84,6 +84,7 @@
   target_link_libraries(
     plplotwxwidgets${LIB_TAG}
     plplotcxx${LIB_TAG} 
+    plplot${LIB_TAG}
     ${wxwidgets_LINK_FLAGS}
     )
 
