Description: Fix cmake D language support 
Author: Andrew Ross <andrewross@users.sourceforge.net>
Origin: upstream
Index: plplot-5.9.9/cmake/modules/language_support/cmake/CMakeDetermineDCompiler.cmake
===================================================================
--- plplot-5.9.9.orig/cmake/modules/language_support/cmake/CMakeDetermineDCompiler.cmake	2011-10-13 01:43:01.000000000 +0100
+++ plplot-5.9.9/cmake/modules/language_support/cmake/CMakeDetermineDCompiler.cmake	2013-06-26 21:52:27.309698965 +0100
@@ -99,10 +99,10 @@
 # a different location.
 IF(EXISTS ${CMAKE_SOURCE_DIR}/cmake/modules/language_support/cmake/CMakeDCompiler.cmake.in)
 	CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/cmake/modules/language_support/cmake/CMakeDCompiler.cmake.in 
-               "${CMAKE_PLATFORM_ROOT_BIN}/CMakeDCompiler.cmake" IMMEDIATE)
+               "${CMAKE_PLATFORM_INFO_DIR}/CMakeDCompiler.cmake" IMMEDIATE)
 ELSE(EXISTS ${CMAKE_SOURCE_DIR}/cmake/modules/language_support/cmake/CMakeDCompiler.cmake.in)
 	CONFIGURE_FILE(${CMAKE_ROOT}/Modules/CMakeDCompiler.cmake.in 
-               "${CMAKE_PLATFORM_ROOT_BIN}/CMakeDCompiler.cmake" IMMEDIATE)
+               "${CMAKE_PLATFORM_INFO_DIR}/CMakeDCompiler.cmake" IMMEDIATE)
 ENDIF(EXISTS ${CMAKE_SOURCE_DIR}/cmake/modules/language_support/cmake/CMakeDCompiler.cmake.in)
 
 MARK_AS_ADVANCED(CMAKE_AR)
