Description: Proper use of the rmpath function
Author: Rafael Laboissiere <rafael@laboissiere.net>
Last-Update: 2012-03-31

Index: plplot-5.9.10/bindings/octave/PLplot/toggle_plplot_use.m
===================================================================
--- plplot-5.9.10.orig/bindings/octave/PLplot/toggle_plplot_use.m	2013-10-01 13:05:57.927300532 +0100
+++ plplot-5.9.10/bindings/octave/PLplot/toggle_plplot_use.m	2013-10-01 13:05:57.915300532 +0100
@@ -39,9 +39,7 @@
 use_plplot_path = plplot_octave_path;
 plplot_path_to_remove = char(strsplit(genpath(use_plplot_path),pathsep));
 for use_plplot_i=1:size(plplot_path_to_remove)(1)
-  if (findstr(path,deblank(plplot_path_to_remove(use_plplot_i,:))) > 0)
-    rmpath(deblank(plplot_path_to_remove(use_plplot_i,:)));
-  endif
+  rmpath(plplot_path_to_remove{use_plplot_i});
 endfor
 
 if (strcmp (use_plplot_state, "on"))
