--- docbook2texixml-orig	Thu Nov 30 13:30:20 2000
+++ docbook2texixml	Mon Nov 27 14:22:22 2000
@@ -940,7 +940,8 @@
 sub dbtr_funcsynopsis {
 
     my $node = shift;
-
+    my $paramdef_start = 1;
+    
     node_map_ordered_children($node,
         {
             'elem:funcsynopsisinfo' => \&tr_example,
@@ -980,6 +981,12 @@
 			    $xw->characters('...');
 			},
                         'elem:paramdef' => sub {
+  			    if ($paramdef_start) {
+			      $paramdef_start = 0;
+			    }
+			    else {
+			      $xw->characters(', ')
+			    }
 			    $xw->characters(&get_cdata);
 			},
                         'whitespace' => sub {},
